tests/nat/conftest.py

```diff
@@ -0,0 +1,227 @@
+    duthost.command("sonic-clear nat translations")
+
+
+@pytest.fixture(params=["loopback", "port_in_lag"], scope='module')
+def setup_test_env(request, ptfhost, duthost, testbed):
+    """
+    setup testbed's environment for CT run
+    :param request: pytest request object
+    :param ptfhost: PTF host object
+    :param duthost: DUT host object
+    :param testbed: Testbed object
+    :return: interface type and testbed setup info
+    """
+    # Check if topology is supported
+    if testbed['topo']['name'] not in SUPPORTED_TOPO:
+        pytest.skip('Unsupported topology')
+    interface_type = request.param
+    interfaces_nat_zone = {}
+    portchannels_port_indices = []
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
+    router_mac = duthost.setup()['ansible_facts']['ansible_Ethernet0']['macaddress']
+    config_port_indices = cfg_facts['port_index_map']
+    config_portchannels = cfg_facts.get('PORTCHANNEL', {})
+    config_ports = cfg_facts['PORT']
+    ptf_ports_available_in_topo = ptfhost.host.options['variable_manager'].extra_vars.get("ifaces_map")
+    # Get outer port indices
+    for port_id in config_portchannels.keys():
+        port = config_portchannels[port_id]['members'][0]
+        portchannels_port_indices.append(config_port_indices[port])
+    ports = [p for p in config_ports
+             if config_port_indices[p] in ptf_ports_available_in_topo and config_ports[p].get('admin_status', 'down') == 'up']
+    vlan_port_indices = [config_port_indices[p] for p in ports]
+    dut_rifs_in_topo_t0 = [el[8:] for el in duthost.setup()['ansible_facts'].keys()
+                           if 'PortCh' in el or 'Loop' in el or 'Vlan' in el]
+    dut_rifs_in_topo_t0 = sorted(dut_rifs_in_topo_t0, reverse=True)
+    inner_zone_interfaces = dut_rifs_in_topo_t0[0]
+    outer_zone_interfaces = dut_rifs_in_topo_t0[1:]
+    for rif in dut_rifs_in_topo_t0:
+        interfaces_nat_zone[rif] = {'interface_name': rif,
+                                    "global_interface_name": '{}_INTERFACE'.format(
+                                        (rif.encode()).translate(None, '0123456789').upper())
+                                    }
+        if rif in inner_zone_interfaces:
+            interfaces_nat_zone[rif]['zone_id'] = 0
+        elif rif in outer_zone_interfaces:
+            interfaces_nat_zone[rif]['zone_id'] = 1
+    indices_to_ports_config = dict((v, k) for k, v in config_port_indices.iteritems())
+    if interface_type == "port_in_lag":
+        outer_zone_interfaces = [outer_zone_interfaces[-2], outer_zone_interfaces[-3]]
+        public_ip = SETUP_CONF["port_in_lag"]["vrf"]["red"]["gw"]
+        second_public_ip = SETUP_CONF["port_in_lag"]["vrf"]["green"]["gw"]
+    else:
+        public_ip = duthost.setup()['ansible_facts']['ansible_Loopback0']['ipv4']['address']
+        second_public_ip = duthost.setup()['ansible_facts']['ansible_Loopback0']['ipv4']['address']
+
```

Comment:
Already have this infomation in  cfg_facts ?

Reply:
Thanks, will be updated with next commit